From 1326aca9622f740d4b8e2b14c511c69291056736 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Thu, 21 Mar 2024 11:04:25 +0100
Subject: [PATCH] media: verisilicon: h1: add h264 rotation support

Add support of input image rotation before encoding.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Change-Id: I3eb6708a371b823e3037e9b3c60a2a37c6e72ed1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368373
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/374020
---
 .../platform/verisilicon/hantro_h1_h264_enc.c | 24 +------------------
 1 file changed, 1 insertion(+), 23 deletions(-)

--- a/drivers/media/platform/verisilicon/hantro_h1_h264_enc.c
+++ b/drivers/media/platform/verisilicon/hantro_h1_h264_enc.c
@@ -221,28 +221,6 @@ static void hantro_h1_h264_enc_set_param
 	}
 }
 
-static void hantro_h1_h264_enc_set_src_img_ctrl(struct hantro_dev *vpu,
-						struct hantro_ctx *ctx)
-{
-	u32 overfill_r, overfill_b;
-	u32 reg;
-
-	/*
-	 * The format width and height are already macroblock aligned
-	 * by .vidioc_s_fmt_vid_cap_mplane() callback. Destination
-	 * format width and height can be further modified by
-	 * .vidioc_s_selection(), and the width is 4-aligned.
-	 */
-	overfill_r = ctx->src_fmt.width - ctx->dst_fmt.width;
-	overfill_b = ctx->src_fmt.height - ctx->dst_fmt.height;
-
-	reg = H1_REG_IN_IMG_CTRL_ROW_LEN(ctx->src_fmt.width)
-		| H1_REG_IN_IMG_CTRL_OVRFLR_D4(overfill_r / 4)
-		| H1_REG_IN_IMG_CTRL_OVRFLB(overfill_b)
-		| H1_REG_IN_IMG_CTRL_FMT(ctx->vpu_src_fmt->enc_fmt);
-	vepu_write_relaxed(vpu, reg, H1_REG_IN_IMG_CTRL);
-}
-
 static void hantro_h1_h264_enc_set_buffers(struct hantro_dev *vpu, struct hantro_ctx *ctx,
 					   struct v4l2_ctrl_h264_encode_params *params)
 {
@@ -395,7 +373,7 @@ int hantro_h1_h264_enc_run(struct hantro
 	vepu_write_relaxed(vpu, H1_REG_ENC_CTRL_ENC_MODE_H264, H1_REG_ENC_CTRL);
 
 	hantro_h1_h264_enc_set_params(vpu, ctx, params, rc->qp);
-	hantro_h1_h264_enc_set_src_img_ctrl(vpu, ctx);
+	hantro_h1_set_src_img_ctrl(vpu, ctx);
 	hantro_h1_h264_enc_set_buffers(vpu, ctx, params);
 
 	hantro_h1_set_color_conv(vpu, ctx);
